[Merged by Bors] - chore(algebra/algebra): reduce imports for `algebra.algebra.tower`

Vierkantor opened the pull request:
This splits `algebra.algebra.tower` into two files: `algebra.algebra.tower.basic` and `algebra.algebra.subalgebra.tower`.

The motivation behind this PR is that `algebra.algebra.tower` is a relatively fundamental and popular file to import (especially since it used to contain `is_scalar_tower`), but its import tree also includes relatively complicated files such as `ring_theory.ideal.operations`. By splitting off these less popular dependencies we can save quite a bit of complexity in the import graph.

---
<!-- The text above the `---` will become the commit message when your
PR is merged. Please leave a blank newline before the `---`, otherwise
GitHub will format the text above it as a title.

To indicate co-authors, include lines at the bottom of the commit message 
(that is, before the `---`) using the following format:

Co-authored-by: Author Name <author@email.com>

Any other comments you want to keep out of the PR commit should go
below the `---`, and placed outside this HTML comment, or else they
will be invisible to reviewers.

If this PR depends on other PRs, please list them below this comment,
using the following format:
- [ ] depends on: #abc [optional extra text]
- [ ] depends on: #xyz [optional extra text]
-->

[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/from-referrer/)

Vierkantor committed:
chore(algebra/algebra): reduce imports for `algebra.algebra.tower`

This splits `algebra.algebra.tower` into three files: `algebra.algebra.tower.basic`, `algebra.algebra.tower.bilinear` and `algebra.algebra.subalgebra.tower`

Vierkantor committed:
Fix imports

Vierkantor committed:
Fix duplicate definition (already lives in `algebra.subalgebra.tower`)

Vierkantor committed:
Fix imports

sgouezel commented on src/algebra/algebra/tower/basic.lean:
```diff
@@ -1,11 +1,10 @@
 /-
```
After the rename, this file is alone in the directory `src/algebra/algebra/tower`. Are you planning to add other files to this directory in another refactor? Otherwise, you might as well keep its current name.

Vierkantor replied:
This folder used to have two other files that got shuffled around over the course of the PR. So indeed might as well be moved back now.

Vierkantor committed:
Move `algebra.algebra.tower.basic` back to `algebra.algebra.tower`

This rename is unneeded now that the other files in that folder have been moved out.
